packages/multichain/src/middlewares/MultichainSubscriptionManager.ts

```diff
@@ -0,0 +1,162 @@
+import { toHex } from '@metamask/controller-utils';
+import createSubscriptionManager from '@metamask/eth-json-rpc-filters/subscriptionManager';
+import type { NetworkController } from '@metamask/network-controller';
+import SafeEventEmitter from '@metamask/safe-event-emitter';
+import type { CaipChainId, Hex } from '@metamask/utils';
+import { parseCaipChainId } from '@metamask/utils';
+
+import type { ExternalScopeString } from '../scope/types';
+import type { ExtendedJsonRpcMiddleware } from './MultichainMiddlewareManager';
+
+export type SubscriptionManager = {
+  events: SafeEventEmitter;
+  destroy?: () => void;
+  middleware: ExtendedJsonRpcMiddleware;
+};
+
+type SubscriptionNotificationEvent = {
+  jsonrpc: '2.0';
+  method: 'eth_subscription';
+  params: {
+    subscription: Hex;
+    result: unknown;
+  };
+};
+
+type SubscriptionKey = {
+  scope: ExternalScopeString;
+  origin: string;
+  tabId?: number;
+};
+type SubscriptionEntry = SubscriptionKey & {
+  subscriptionManager: SubscriptionManager;
+};
+
+type MultichainSubscriptionManagerOptions = {
+  findNetworkClientIdByChainId: NetworkController['findNetworkClientIdByChainId'];
+  getNetworkClientById: NetworkController['getNetworkClientById'];
+};
+
+export class MultichainSubscriptionManager extends SafeEventEmitter {
```

Comment:
What is the purpose of this class and how is it intended to be used?

Reply:
added desc here and jsdoc to the constructor https://github.com/MetaMask/core/pull/4813/commits/8874ebcbc5620fac8f743a2ff351b35483fa411a